lib/router/router.ts

```diff
@@ -120,7 +120,7 @@ export default abstract class Router<T extends Route> {
       // perform a URL update at the end. This gives
       // the user the ability to set the url update
       // method (default is replaceState).
-      let newTransition = new InternalTransition(this, undefined, undefined);
+      let newTransition = new InternalTransition(this, undefined, newState);
       newTransition.queryParamsOnly = true;
```

Comment:
```suggestion
      newTransition.queryParamsOnly = true;
      this.setupContexts(newState, newTransition);
```

Some other tests started breaking with the `undefined` -> `newState` change in L123, because this apparently leaks through Routes which haven't yet been passed through `setupContexts(...)` and thus did not have their `setup(...)` hook called.

Ultimately this causes [`Route#finalizeQueryParamChange`](https://github.com/emberjs/ember.js/blob/8f56dc80620a47aecb602e9f1ca18a1c85d83b9a/packages/%40ember/-internals/routing/lib/system/route.ts#L2547) to fail, as it [expects `route.controller` to be defined](https://github.com/emberjs/ember.js/blob/8f56dc80620a47aecb602e9f1ca18a1c85d83b9a/packages/%40ember/-internals/routing/lib/system/route.ts#L2572).